include/zephyr/net/offloaded_netdev.h

```diff
@@ -29,6 +29,19 @@ extern "C" {
  * @{
  */
 
+/** Types of offloaded netdev L2 */
+enum offloaded_net_types {
+	/** Networking stack offloaded (default case).
+	 *
+	 * Note: To avoid changing drivers, using special value 0 instead
+	 * of BIT(0) for this type.
+	 */
+	L2_OFFLOADED_NET_TYPE_ETHERNET		= 0,
```

Comment:
0 (default value) should be rather reserved for `UNKNOWN` type, we don't want for example LTE offloaded devices to be represented as Ethernet. Actually, I'm not really sure if we have any offloaded Ethernet devices in the tree.